packages/fetchai/connections/p2p_libp2p/connection.py

```diff
@@ -202,11 +210,12 @@ def __init__(
 
         # log file
         self.log_file = log_file if log_file is not None else LIBP2P_NODE_LOG_FILE
-        self.log_file = os.path.join(os.path.abspath(os.getcwd()), self.log_file)
-
+        if not Path(self.log_file).is_absolute():
+            self.log_file = os.path.abspath(os.path.join(data_dir, self.log_file))
```

Comment:
@DavidMinarsch I noticed this:

```python
import os
p1 = "/some/abs/path
p2 = "/another/abs/path"
p3 = os.path.join(p1, p2)  # "/another/abs/path"
```
That is, concatenation of two absolute paths gives the latter one.

We could update this and other parts where we do a similar operation; would simplify the code (at the cost of worse readability!)

Reply:
I fixed it now I think (data_dir is absolute, self.log_file is not as per check)

Reply:
`data_dir` can be also a relative path wrt working directory. The concatenation would give the latter anyway if the latter is absolute

Reply:
The latter cannot be absolute as per check!